src/components/TextField/TextField.tsx

```diff
@@ -56,94 +24,65 @@ export type Props = {
-      <div css={iconWrapperStyle({ iconPosition })}>{children}</div>
-    );
+  const IconWrapper: FC<{ iconPosition: 'left' | 'right' }> = ({ children, iconPosition }) => (
+    <div css={iconWrapperStyle({ iconPosition })}>{children}</div>
+  );
 
-    return (
-      <React.Fragment>
-        <div css={{ display: 'flex', alignItems: 'center', position: 'relative' }}>
-          <div css={wrapperStyle({ dark, locked, disabled, status, lean, styleType })}>
-            <div css={textFieldStyle({ size, label, leftIcon })}>
-              {leftIcon && <IconWrapper iconPosition={'left'}>{leftIcon}</IconWrapper>}
-              <div css={{ width: '100% ' }}>
-                <input
-                  css={inputStyle({ label, placeholder, size, dark })}
-                  placeholder={
-                    !label && placeholder ? `${placeholder} ${required ? '*' : ''}` : label
-                  }
-                  required={required}
-                  id={id}
-                  disabled={disabled || locked}
-                  {...rest}
-                  ref={ref}
-                />
-                {label && (
-                  <Label
-                    size={size}
-                    htmlFor={id}
-                    label={label}
-                    required={required}
-                    animateToTop={Boolean(rest.value)}
-                  />
-                )}
-              </div>
-              {rightIcon && !locked && (
-                <IconWrapper iconPosition={'right'}>{rightIcon}</IconWrapper>
-              )}
-              {locked && (
-                <IconWrapper iconPosition={'right'}>
-                  <Icon
-                    name="lock"
-                    size={size === 'md' ? 20 : 16}
-                    color={theme.utils.getColor('lightGray', 500)}
-                  />
-                </IconWrapper>
-              )}
-            </div>
-          </div>
+  return (
+    <React.Fragment>
+      <TextInputWrapper {...props}>
+        {leftIcon && <IconWrapper iconPosition={'left'}>{leftIcon}</IconWrapper>}
+        <div css={{ width: '100% ' }}>
+          <input
+            css={inputStyle({ label, placeholder, size, dark })}
+            placeholder={!label && placeholder ? `${placeholder} ${required ? '*' : ''}` : label}
+            required={required}
+            id={id}
+            disabled={disabled || locked}
+            {...rest}
+            ref={ref}
+          />
+          {label && (
+            <Label
+              size={size}
+              htmlFor={id}
+              label={label}
+              required={required}
+              animateToTop={Boolean(rest.value)}
+            />
+          )}
         </div>
-        {hintMsg && status !== 'normal' && (
-          <div css={errorMsgStyle({ status })}>{hintMessageToShow}</div>
+        {rightIcon && !locked && <IconWrapper iconPosition={'right'}>{rightIcon}</IconWrapper>}
+        {locked && (
```

Comment:
You can move this condition inside IconWrapper's body.

example: 

```
 const IconWrapper: FC<{ iconPosition: 'left' | 'right' }> = ({ children, iconPosition }) => (
    locked && <div css={iconWrapperStyle({ iconPosition })}>{children}</div>
  );
```

Reply:
actually this makes it harder for me to understand 😄 , i dont fully agree with this `shouldDisplay` functionality